### PR TITLE
JSON parser commit

### PR DESCRIPTION
**Pull request by tavisca-jayesh**
@apalsapure  Please review the code

### CODE REVIEW

File: 01 Operator Overloading/OperatorOverloading.DBL/APIFetchFile.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace OperatorOverloading.DBL
+{
+    class APIFetchFile
+    {
+        public static string getJSON()
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/APIFetchFile.cs]: Naming convension

File: 01 Operator Overloading/OperatorOverloading.DBL/APIFetchFile.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            string path = "C:/Users/jyeola/Desktop/conversion_rates.txt";
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/APIFetchFile.cs]: Read it from configuration file

File: 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OperatorOverloading.DBL
+{
+    public class CurrencyConverter : ICurrencyConverter
+    {
+        public double GetConversionRate(string source, string target)
+        {
+            if (this == null)
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs]: `this` can not be null

File: 01 Operator Overloading/IntegrationTest/CurrencyConversionTest.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            var money1 = new Money(null);
+            var money2 = money1.Convert("INR");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(NullReferenceException))]
+        public void MoneyObjectNull()
+        {
+            Money money1 = null;
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/IntegrationTest/CurrencyConversionTest.cs]: Useless test case

File: 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            }
+            //API Fetch
+            string json = APIFetchFile.getJSON();
+            //JSON Parse
+            string[] conversionStrings = JSONParser.parseJSON(json);
+
+            //Calculate Conversion Rate    
+            foreach (string i in conversionStrings)
+            {
+                if (i.Contains(target.ToUpper()))
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs]: do ToUpper before entering the loop.

File: 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                throw new System.Exception(Messages.CurrencyInvalid);
+            }
+            //API Fetch
+            string json = APIFetchFile.getJSON();
+            //JSON Parse
+            string[] conversionStrings = JSONParser.parseJSON(json);
+
+            //Calculate Conversion Rate    
+            foreach (string i in conversionStrings)
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs]: What is `i`?

File: 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                throw new NullReferenceException();
+            }
+            if (string.IsNullOrWhiteSpace(source) || string.IsNullOrWhiteSpace(target) || source.Length != 3 || target.Length != 3)
+            {
+                throw new System.Exception(Messages.CurrencyInvalid);
+            }
+            //API Fetch
+            string json = APIFetchFile.getJSON();
+            //JSON Parse
+            string[] conversionStrings = JSONParser.parseJSON(json);
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs]: Use `Dictonery`

File: 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+            //Calculate Conversion Rate    
+            foreach (string i in conversionStrings)
+            {
+                if (i.Contains(target.ToUpper()))
+                {
+                    string[] convertRate = i.Split(':');
+                    double amount;
+                    if (double.TryParse(convertRate[1], out amount) == false || amount <0 || double.IsPositiveInfinity(amount) || amount == double.MaxValue)
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/CurrencyConverter.cs]: What will happen if storage returns `INR-USD`

File: 01 Operator Overloading/OperatorOverloading.DBL/JSONParser.cs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OperatorOverloading.DBL
+{
+    class JSONParser
+    {
+        public static string[] parseJSON(string json)
```
Comment (tavisca-jayesh) [inline 01 Operator Overloading/OperatorOverloading.DBL/JSONParser.cs]: Naming convension

**Comment by apalsapure** [comment]
Your score is 1.
- Naming convention issue makes score to zero
- But you have implemented and validated code correctly hence 1

Fix the code review comment and then merge your code.

**Pull request merged**